### ISSUE TITLE
Deleting an image and reuploading new one with same name shows the old one

### ISSUE THREAD

**Issue by freshface** [statamic/cms#2751]
## Bug Description

When i delete an image in a folder and upload a new one with the same name it always shows the old one.
Even after a refresh

## Environment

**Statamic version:** 3.0.20

**PHP version:** 7.3

**Install method (choose one):**
- Fresh install from `statamic/statamic`

**Comment by robdekort** [comment]
Could it be cached by your browser, what happens when you empty your browser cache?

**Comment by freshface** [comment]
Hard refresh, same issue.
But will try again to be sure.

**Comment by robdekort** [comment]
Do you auto generate asset presets? Maybe they currently don't get deleted when you delete an asset. Just thinking out loud here :-). You could try clearing those as well.

**Comment by damcclean** [comment]
It might be worth also clearing your Glide cache after you've deleted an entry to see if that's the cause. `php please glide:clear`

**Comment by robdekort** [comment]
Yeah that's what I meant. Just confirmed it's the glide cache. So the issue is that when you delete an asset the glide cache for that asset doesn't get's purged.

**Comment by freshface** [comment]
Thx @robdekort for checking!

**Comment by robdekort** [comment]
Go away github-actions bot!

**Comment by EikeBierman** [comment]
Do you have a workaroud or something for this problem ?

**Comment by robdekort** [comment]
You have to clear the Glide cache and possibly your browser cache. You could also manually delete the generated presets for that asset from the glide cache so you keep others.

**Comment by damcclean** [comment]
When you say it shows the old one, are you talking about in the image previews in the Asset browser?

**Comment by robdekort** [comment]
I assume so cuz I noticed it as well. It also happens on the frontend when you use the glide cache.

**Comment by EikeBierman** [comment]
ok after a hard browser clear and php please glide:clear it worked. But I can't tell my client to use php please glide:clear. could i disable it?

**Comment by robdekort** [comment]
You could disable the glide cache, but that could drastically decrease your sites performance. I'd instruct your client not to upload assets with the same name, for now.

Another thing you could do is give the client access to the caching utilities in the CP. They can clear the glide cache with the press of a button.

**Comment by stuartcusackie** [comment]
I am having similar trouble with a same-name file, and no matter what I do, the old image size keeps coming back:
- Use statamic utilities to clear all caches
- Use php please glide:clear
- Manually delete /img/ folder on server
- Cleared my browser cache.

It's possibly a cloudflare caching thing, and I have partially cleared that cache (I can't fully clear it at the moment).

Can't figure it out...

**Comment by freshface** [comment]
bump

**Comment by twd3** [comment]
This really needs to be fixed. It's a big deal when a client goes around in circles simply trying to replace an image. I just also cleared all caches and the image won't update. Also, to instruct a client to do all this clearing stuff and try again is NOT a working solution. It's a dang image replacement. It should be easy and yes they almost always use the same filename because they thought of a good one when they output the image so they don't want to think of another one.

**Comment by robdekort** [comment]
Please consider the tone of your replies. I get that sometimes bugs can be frustrating. I've been there myself. But this won't help. There's a lot of issues to fix and at some point they'll definitely get to this one. Maybe it can get priority if you're lucky. Thumbing the original post up helps as they devs usually sort by thumbed up issues to get a view of what needs priority.

**Comment by twd3** [comment]
I hear you but I also have the right to be annoyed here. This was reported March 2020. This is a paid product. This is an issue that should get priority over adding some new feature because it has an effect on end users.

**Comment by jasonvarga** [comment]
I can fix the Glide issue (I have it ready locally) but there will still be the problem that the browser will want to cache it incredibly hard, which we can't do a whole lot about. Aside, of course, from instructing the browser to not cache it, but that's an even worse idea.

Once the browser gets the image at that URL, it'll remember it for a year unless you clear your browser cache. Images generally won't ever change, so it's generally an accepted practice to cache them for a long time.

Point is, a "real" fix for this is probably quite significant.

For the moment I'd suggest telling your client not to do that.

As for the delay, if there's an issue that is causing you a lot of grief, feel free to send a support request. https://statamic.com/support

**Comment by jasonvarga** [comment]
Closing since we now have the reupload asset feature.

**Comment by parpar78** [comment]
Not sure if i should start a new issue, but the same issue pointed out above is still happening for me even when using 'reupload'

I should also add a disclaimer here, i am not familliar with statamic but i am taking over while my collegue is on vacation, and am not sure where else i can find help with this.

It is an image that is used in several emails, so using a different URL is not an option.

So i guess my question is if anyone has experience with this and how to work around?
I already cleared the Image Cache through utilities.

Situation is now as follows:
- When downloading it downloads the new version, so that works
- when visiting in browser it shows old version (even in incognito mode and other devices so that should rule out browser cache right?)
- Shows old version in the assets manager aswell.

I also tried:
- delete old version
- upload new under different name
- rename to old name
But as soon as i rename it shows the old one again...

Sorry for the chaotic comment, i hope it is understandable and someone can give me some pointers.
